### PR TITLE
Update Stackdriver Trace exporter to use V2 API.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    SpanName spanName =
+        SpanName.newBuilder().setProject(projectId).setTrace(traceIdHex).setSpan(spanIdHex).build();
+    Span.Builder spanBuilder =
+        Span.newBuilder()
+            .setNameWithSpanName(spanName)
+            .setSpanId(encodeSpanId(context.getSpanId()))
+            .setDisplayName(toTruncatableStringProto(spanData.getName()))
+            .setStartTime(toTimestampProto(spanData.getStartTimestamp()))
+            .setAttributes(toAttributesProto(spanData.getAttributes()))
+            .setTimeEvents(
```
Comment (adriancole) [inline exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java]: time events is new right?

File: exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    return attributeValueBuilder.build();
+  }
+
+  @Override
+  public void export(Collection<SpanData> spanDataList) {
+    List<Span> spans = new ArrayList<>(spanDataList.size());
+    for (SpanData spanData : spanDataList) {
+      spans.add(generateSpan(spanData));
+    }
+    traceServiceClient.batchWriteSpans(projectName, spans);
```
Comment (adriancole) [inline exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java]: maybe add a comment what this implies? Ex is this async

File: exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+  private static Status toStatusProto(io.opencensus.trace.Status status) {
+    Status.Builder statusBuilder = Status.newBuilder().setCode(status.getCanonicalCode().value());
+    if (status.getDescription() != null) {
+      statusBuilder.setMessage(status.getDescription());
+    }
+    return statusBuilder.build();
+  }
+
+  private static TruncatableString toTruncatableStringProto(String string) {
```
Comment (g-easy) [inline exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java]: I think this needs to take a max len arg, and setTruncatedByteCount

File: exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+    return timeEventsBuilder.build();
+  }
+
+  private static TimeEvent toTimeAnnotationProto(TimedEvent<Annotation> timedEvent) {
+    TimeEvent.Builder timeEventBuilder =
+        TimeEvent.newBuilder().setTime(toTimestampProto(timedEvent.getTimestamp()));
+    Annotation annotation = timedEvent.getEvent();
+    timeEventBuilder.setAnnotation(
+        TimeEvent.Annotation.newBuilder()
+            .setDescription(toTruncatableStringProto(annotation.getDescription()))
```
Comment (g-easy) [inline exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java]: Truncate to 256 bytes

File: exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+    SpanContext context = spanData.getContext();
+    final String traceIdHex = encodeTraceId(context.getTraceId());
+    final String spanIdHex = encodeSpanId(context.getSpanId());
+    SpanName spanName =
+        SpanName.newBuilder().setProject(projectId).setTrace(traceIdHex).setSpan(spanIdHex).build();
+    Span.Builder spanBuilder =
+        Span.newBuilder()
+            .setNameWithSpanName(spanName)
+            .setSpanId(encodeSpanId(context.getSpanId()))
+            .setDisplayName(toTruncatableStringProto(spanData.getName()))
```
Comment (g-easy) [inline exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java]: Truncate to 128.

File: exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+  }
+
+  private static AttributeValue toAttributeValueProto(
+      io.opencensus.trace.AttributeValue attributeValue) {
+    final AttributeValue.Builder attributeValueBuilder = AttributeValue.newBuilder();
+    attributeValue.match(
+        new Function<String, Void>() {
+          @Override
+          public Void apply(String stringValue) {
+            attributeValueBuilder.setStringValue(toTruncatableStringProto(stringValue));
```
Comment (g-easy) [inline exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java]: Truncate to 256

File: exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java (hunk 1 above)
Comment (bogdandrutu) [reply exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java]: yes.

File: exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java (hunk 5 above)
Comment (bogdandrutu) [reply exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java]: https://github.com/census-instrumentation/opencensus-java/issues/813

File: exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java (hunk 4 above)
Comment (bogdandrutu) [reply exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java]: https://github.com/census-instrumentation/opencensus-java/issues/813

File: exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java (hunk 3 above)
Comment (bogdandrutu) [reply exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java]: We will add truncation in the future.

File: exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java (hunk 6 above)
Comment (bogdandrutu) [reply exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java]: https://github.com/census-instrumentation/opencensus-java/issues/813

File: exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java (hunk 2 above)
Comment (bogdandrutu) [reply exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java]: done.

**Pull request merged**